frontend/src/pages/CompareV2.tsx

```diff
@@ -557,6 +531,7 @@ function MetricsDropdown(props: MetricsDropdownProps) {
 function CompareV2(props: PageProps) {
   const { updateBanner, updateToolbar } = props;
 
+  const namespace: string | undefined = useContext(NamespaceContext);
```

Comment:
I'm not certain that this fix is actually appropriate? The previous method seemed to be getting the `namespace` from the selected run - but should these already all be on the same `namespace`, as the user is viewing them on the Run Comparison page? Please let me know what you think of this fix @zijianjoy.

Reply:
We can move the `namespace` to be a props field which looks like the following:

```
namespace?: string
```

Then in `EnhancedCompareV2`, we can call `const namespace = useContext(NamespaceContext);` and pass the namespace as props when calling `<CompareV2>.

Reply:
Ok, done here: https://github.com/kubeflow/pipelines/pull/8036/commits/4627a1f991796dfc30e8ff632f184ab07fd13038. I did not add or modify any tests with this change as the only behavior update is that the passed `namespace` parameter would be used when fetching the HTML or Markdown file, and that is already tested in `MetricsDropdown.test.tsx`.